python_modules/dagster/dagster/core/execution.py

```diff
@@ -584,17 +516,41 @@ def execute_pipeline(
     '''
 
     check.inst_param(pipeline, 'pipeline', PipelineDefinition)
-    check.opt_inst_param(environment, 'environment', (dict, config.Environment))
+    check.opt_dict_param(environment, 'environment')
     check.bool_param(throw_on_error, 'throw_on_error')
     check.opt_inst_param(reentrant_info, 'reentrant_info', ReentrantInfo)
 
-    check_environment(pipeline, environment)
+    typed_environment = get_typed_environment(pipeline, environment)
+
+    execution_graph = ExecutionGraph.from_pipeline(pipeline)
```

Comment:
This is a parallel code path to ll. 534-540, we should just hand off from the public api to `execute_reentrant_pipeline` instead